### PR TITLE
Bug fix/check analyzers in pipeline

### PR DESCRIPTION
**Pull request by Dronplane**
### Scope & Purpose
Pipeline analyzer could hold arbitrary analyzers in pipe. And if one piped analyzer emits output non compatible with next analyzer in the pipe - crash could occur. PR adds validation procedure to verify type acceptance between pipe members.

- [x] :hankey: Bugfix (requires CHANGELOG entry)
- [ ] :pizza: New feature (requires CHANGELOG entry, feature documentation and release notes)
- [ ] :fire: Performance improvement
- [ ] :hammer: Refactoring/simplification
- [x] :book: CHANGELOG entry made

#### Backports:

- [ ] No backports required
- [x] Backports required for: 3.8

### Testing & Verification
- [ ] This change is a trivial rework / code cleanup without any test coverage.
- [x] The behavior in this PR was *manually tested*
- [ ] This change is already covered by existing tests, such as *(please describe tests)*.
- [x] This PR adds tests that were used to verify all changes:
  - [ ] Added new C++ **Unit tests**
  - [x] Added new  shell_client_aql test
  - [ ] Added new **resilience tests** (only if the feature is impacted by failovers)
- [ ] There are tests in an external testing repository:
- [ ] I ensured this code runs with ASan / TSan or other static verification tools

Link to Jenkins PR run:

### CODE REVIEW

File: 3rdParty/iresearch/core/analysis/pipeline_token_stream.hpp [hunk 1]
```diff
@@ -92,6 +102,10 @@ class pipeline_token_stream final
     uint32_t data_end{ 0 };
     uint32_t pos{ std::numeric_limits<uint32_t>::max() };
 
+    irs::analysis::analyzer::ptr get_stream() const noexcept {
+      return analyzer;
+    }
```
Comment (gnusi) [inline 3rdParty/iresearch/core/analysis/pipeline_token_stream.hpp:107]:
```suggestion
irs::analysis::analyzer& get_stream() const noexcept {
  assert(analyzer);
  return *analyzer;
}

File: 3rdParty/iresearch/core/analysis/pipeline_token_stream.hpp [hunk 2]
```diff
@@ -53,6 +55,14 @@ class pipeline_token_stream final
   virtual bool next() override;
   virtual bool reset(const string_ref& data) override;
 
+  //////////////////////////////////////////////////////////////////////////////
+  /// @brief calls visitor on pipeline members in respective order. Visiting is 
+  /// interrupted on first visitor returning false.
+  /// @param visitor visitor to call
+  /// @return true if all visits returned true, false otherwise
+  //////////////////////////////////////////////////////////////////////////////
+  bool visit_members(const std::function<bool(const irs::analysis::analyzer::ptr)>& visitor) const;
```
Comment (gnusi) [inline 3rdParty/iresearch/core/analysis/pipeline_token_stream.hpp:64]: maybe use template-based visitor?

Comment (Dronplane) [reply 3rdParty/iresearch/core/analysis/pipeline_token_stream.hpp:64]: That may create unnecessary complexity  for shared builds. But if we drop shared builds than that will be more flexible of course.

File: 3rdParty/iresearch/core/analysis/pipeline_token_stream.hpp [hunk 3]
```diff
@@ -48,11 +50,43 @@ class pipeline_token_stream final
 
   explicit pipeline_token_stream(options_t&& options);
   virtual attribute* get_mutable(irs::type_info::type_id id) noexcept override {
-    return irs::get_mutable(attrs_, id);
+    auto atr = irs::get_mutable(attrs_, id);
```
Comment (gnusi) [inline 3rdParty/iresearch/core/analysis/pipeline_token_stream.hpp:53]: `atr` -> `attr` :-)

**Comment by Dronplane** [comment]
Tests blue

**Pull request merged**